File: api-xmlsignature/src/tr/gov/tubitak/uekae/esya/api/xmlsignature/document/FileDocument.java
package tr.gov.tubitak.uekae.esya.api.xmlsignature.document;

import tr.gov.tubitak.uekae.esya.api.xmlsignature.DataType;
import tr.gov.tubitak.uekae.esya.api.xmlsignature.XMLSignatureException;

import java.io.File;
import java.io.FileInputStream;
import java.io.InputStream;
import java.net.URLConnection;

/**
 * Document for file.
 *
 * @author ahmety
 * date: May 14, 2009
 */
public class FileDocument extends Document
{
    private File mFile;

    public FileDocument(File aFile) throws XMLSignatureException
    {
        this(aFile, URLConnection.guessContentTypeFromName(aFile.getName()));
    }

    public FileDocument(File aFile, String aMIMEType) throws XMLSignatureException
    {
        this(aFile, aMIMEType, null);
    }

    public FileDocument(File aFile, String aMIMEType, String aEncoding) throws XMLSignatureException
    {
        super(aFile.getPath(), aMIMEType, aEncoding);
        if (!aFile.exists()){
            throw new XMLSignatureException("errors.cantFind", aFile.getAbsolutePath());
        }
        mFile = aFile;
    }

    public String getURI()
    {
        return mFile.getPath();
    }

    public DataType getType()
    {
        return DataType.OCTETSTREAM;
    }

    public InputStream getStream() throws XMLSignatureException
    {
        try {
            return new FileInputStream(mFile);
        } catch (Exception x){
            // shouldn't happen
            throw new XMLSignatureException(x, "I/O error on FileDocument.getStream");
        }
    }

}

File: api-xmlsignature/csharp/src/document/FileDocument.cs
using System;
using System.IO;
using tr.gov.tubitak.uekae.esya.api.common;
using tr.gov.tubitak.uekae.esya.api.xmlsignature.util;

namespace tr.gov.tubitak.uekae.esya.api.xmlsignature.document
{

	using DataType = tr.gov.tubitak.uekae.esya.api.xmlsignature.DataType;
	using XMLSignatureException = tr.gov.tubitak.uekae.esya.api.xmlsignature.XMLSignatureException;


	/// <summary>
	/// Document for file.
	/// 
	/// @author ahmety
	/// date: May 14, 2009
	/// </summary>
	public class FileDocument : Document
	{
	    private FileStream fileStream=null;
        private readonly FileInfo mFile;

        public FileDocument(string filePath, string aMIMEType, string aEncoding)
            : this(new FileInfo(filePath), aMIMEType,aEncoding)
        {
        }

        public FileDocument(string filePath)
            : this(new FileInfo(filePath))
        {
        }

        public FileDocument(string filePath, string aMIMEType)
            : this(new FileInfo(filePath), aMIMEType)
        {
        }

        public FileDocument(FileInfo aFile)
            : this(aFile, EMimeUtil.GetMimeType(aFile.Name))
		{
		}

		public FileDocument(FileInfo aFile, string aMIMEType) : this(aFile, aMIMEType, null)
		{
		}

		public FileDocument(FileInfo aFile, string aMIMEType, string aEncoding) : base(aFile.FullName, aMIMEType, aEncoding)
		{

            if (!aFile.Exists)
			{
				throw new XMLSignatureException("errors.cantFind", aFile.FullName);
			}
			mFile = aFile;
		}

		public override string URI
		{
			get
			{
				return mFile.FullName;
			}
		}

		public override DataType Type
		{
			get
			{
				return DataType.OCTETSTREAM;
			}
		}

		public override Stream Stream
		{
			get
			{
				try
				{
                    fileStream = File.Open(mFile.FullName, FileMode.Open, FileAccess.Read);
                    return fileStream;				    
				}
				catch (Exception x)
				{
					throw new ESYAException("I/O error on FileDocument.getStream", x);
				}
			}
		}

	}

}